crates/cheatnet/tests/starknet/resources.rs

```diff
@@ -0,0 +1,60 @@
+use std::collections::HashMap;
+
+use crate::common::{
+    deploy_contract, felt_selector_from_name, get_contracts, state::create_cheatnet_state,
+};
+use cairo_felt::Felt252;
+use cheatnet::rpc::{call_contract, CallContractOutput, ResourceReport};
+use conversions::StarknetConversions;
+
+#[test]
+fn call_resources_simple() {
+    let mut state = create_cheatnet_state();
+
+    let contract_address = deploy_contract(&mut state, "HelloStarknet", &[]);
+
+    let selector = felt_selector_from_name("increase_balance");
+
+    let output = call_contract(
+        &contract_address,
+        &selector,
+        &[Felt252::from(123)],
+        &mut state,
+    )
+    .unwrap();
+
+    assert!(match output {
+        CallContractOutput::Success {
+            resource_report, ..
+        } => {
+            resource_report
+                == ResourceReport {
+                    gas: 1.26,
+                    steps: 126,
+                    bultins: HashMap::from([("range_check_builtin".to_owned(), 2)]),
+                }
+        }
+        _ => false,
+    });
+}
+
+#[test]
+fn deploy_resources_simple() {
+    let mut state = create_cheatnet_state();
+
+    let contracts = get_contracts();
+
+    let contract_name = "HelloStarknet".to_owned().to_felt252();
+    let class_hash = state.declare(&contract_name, &contracts).unwrap();
```

Comment:
Shouldn't declare return `resource_report` too?

Reply:
Good question tbh. If we return it in deploy maybe we should return it in declare as well. We can always not include them by default in the summary gas report (on forge's side) and add a flag or sth to include costs of declare and deploy transactions

Reply:
It'd require an overhaul of `declare` this should not be in this PR's scope.

Reply:
#636

Reply:
Declaring does not include execution of any cairo code AFAIK